### PR TITLE
Automated release with changelog for the stable version

### PR DESCRIPTION
**Pull request by haasad**
This PR reduces the effort required to add a new release of the stable version to a single action by the maintainer: Pushing a new tag on the master branch. Everything else is automated, including a changelog in the release notes.

Up until now, a maintainer had to manually create a release on github in order to upload a new stable version to conda. Also the manual [changelog](https://github.com/LCA-ActivityBrowser/activity-browser/blob/master/CHANGELOG.md) has not been updated in over a year.

With this PR the release process is now as follows:
- A maintainer pushes a new tag, e.g.: 
```
git tag -a 2.6.8
git push origin 2.6.8
```
- github actions creates a changelog based on all merged and __labeled__ PRs since the last release
- github actions creates a release on github which includes the changelog from the previous step in the release notes
    - this looks something like this: https://github.com/haasad/activity-browser/releases/tag/0.3.9
    - or this: https://github.com/haasad/EcoInventDownLoader/releases/tag/1.4.1
- the conda package is built and uploaded to the bsteubing channels as before

This only works well under the following conditions:
- PR titles show up in the changelog, so they should summarize the change well
- PRs need to be labeled, otherwise they're not included in the changelog
- The PRs show up in changelog category according to the label(s), I defined some categories in [ci/changelog-configuration.json](https://github.com/LCA-ActivityBrowser/activity-browser/compare/master...haasad:automate-release#diff-3f5730440d4aef36cc8968c4cf801346373d09c4240579693b01cb8d26b42778), but they can always be adapted
- Changes directly commited to master won't be included in the changelog (but could be manually edited in afterwards)

I added a pull request template with a check-list to ensure that setting a good title and labeling is not forgotten. It's mainly a responsibility for the maintainers to ensure that this is done.

Of course this still works when labels etc are forgotten, it will just reduce the quality of the changelog.

### CODE REVIEW

**Comment by haasad** [comment]
@bsteubing I think this is a big quality-of-life improvement for the maintenance of the AB, but it's a "big" change, so please let me know if anything is unclear.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (marc-vdm) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: Perhaps we could add a `\<!-- ... -->` block here with explanation of what we think is meant by `ui`, `change` etc? Or am I overthinking this?

File: ci/changelog-configuration.json
Comment (marc-vdm) [inline ci/changelog-configuration.json:3]: Perhaps we could add or replace the existing labels here too;
`type: feature`
`info: UI`
`type: documentation`?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (marc-vdm) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: A `\## Description` header might also be useful.

**Comment by marc-vdm** [comment]
@haasad: @bsteubing and I discussed the PR and think it's a good idea to include this. Could you still have a look at the comments I left previously, and then feel free to merge this!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (marc-vdm) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: A few additional thoughts:

Perhaps we should clarify \- [ ] Update the documentation to contain sub checkboxes:
\- [ ] Documentation for users has been updated (e.g. the [readme](https://github.com/LCA-ActivityBrowser/activity-browser/blob/master/README.md) or [wiki](https://github.com/LCA-ActivityBrowser/activity-browser/wiki)
\- [ ] All functions and classes have docstrings following [PEP 257](https://peps.python.org/pep-0257/)
\- [ ] All functions have type hinting following [PEP 484](https://peps.python.org/pep-0484/)
This could help contributors to make sure their code is well written -even though we still need to do all of the above for most of AB-

We may also want an additional check:
\- [ ] All relevant issues/discussions/projects have been tagged in the description
This will help maintainers keep track of issues to close and progress in general. I think there could be quite a few issues still open that have long since been resolved because we simply missed them during a merge and then forgot about them.

Comment (haasad) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: > Perhaps we could add a \ block here with explanation of what we think is meant by ui, change etc? Or am I overthinking this?

The checkbox for setting the label is in my opinion more of a reminder for the reviewer to make sure that the PR shows up in the changelog, so the reviewer can always clarify or change the label if the person opening the PR doesn't know what label to use. In order not to overload the PR template I would leave it as it is.

> Perhaps we should clarify - [ ] Update the documentation to contain sub checkboxes:
> - [ ] For features: Documentation for users has been updated (e.g. the [readme](https://github.com/LCA-ActivityBrowser/activity-browser/blob/master/README.md) or [wiki](https://github.com/LCA-ActivityBrowser/activity-browser/wiki))
> - [ ] All functions and classes have docstrings following [PEP 257](https://peps.python.org/pep-0257/)
> - [ ] All functions have type hinting following [PEP 484](https://peps.python.org/pep-0484/)
> This could help contributors to make sure their code is well written -even though we still need to do all of the above for most of AB-

While I agree that this would be nice to have, I'm afraid this could potentially scare away contributors. I guess that's something we could add once the AB code base is in a state where we can realistically expect the same for no contributions. But asking for type hints when there's not even an example in the AB code base how to do that is a bit much to ask imho.

Also we can always ask for improvements in the review, I wouldn't make any of this a hard requirement in the PR template.

> We may also want an additional check:
> - [ ] All relevant issues/discussions/projects have been tagged in the description
> This will help maintainers keep track of issues to close and progress in general. I think there could be quite a few issues still open that have long since been resolved because we simply missed them during a merge and then forgot about them.

There is already `- [ ] Link this PR to related issues`, is that not clear enough?

File: ci/changelog-configuration.json
Comment (haasad) [reply ci/changelog-configuration.json:3]: Afaik the labels on github are shared between Issues and PRs, I'm not sure how or what for those `type: ` and `info: ` labels are used :man_shrugging: 
My guess is that they were used in an effort to categorize open issues some time ago. But no objection from my side to consolidate or clean up.

Also the labels/categories I used here for the PR template / changelog are totally up for debate. It's really just a first suggestion of what I thought could be useful, but not hard preference whatsoever.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (haasad) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Could you please clarify what you mean here? I'm afraid I don't understand

**Comment by haasad** [comment]
@marc-vdm Sorry for the long time it took me to reply, I've added some comments now. And thanks a lot for the detailed review :handshake:

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (haasad) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: Regarding the first point; do you think simply adding in brackets what the abbreviations (ui, ci etc.) mean would already be enough?

Comment (marc-vdm) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: Alright, I think you're right, this may indeed be too much. I got a bit carried away there. Lets keep it simple and we can indeed always ask for improvements during review.

>There is already `- [ ] Link this PR to related issues`, is that not clear enough?

It is, yes, I missed that.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (marc-vdm) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: I meant that we have an 'area' in the template for contributor to provide a description of what the PR is supposed to achieve.

**Comment by marc-vdm** [comment]
@haasad No worries, we're all busy sometimes. Just merge when you're ready to go 👍

**Comment by marc-vdm** [comment]
@bsteubing I think there were still a few very minor things for @haasad to solve, though we're not specifically in a hurry, as long as this is merged before the next stable release everything should work.

**Comment by haasad** [comment]
I'll take a look this week and will merge it when ready

**Comment by marc-vdm** [comment]
@haasad May i ping you on this again?

**Comment by bsteubing** [comment]
This PR is not forgotten, I think we should accept it, but I will let @Zoophobus do that as soon as he understands / agrees with this approach.

File: ci/changelog-configuration.json
Comment (Zoophobus) [reply ci/changelog-configuration.json:3]: @haasad @marc-vdm
First of all I don't know too much about the standards when it comes to github actions. This is my first exposure to them, the other projects I have been involved with have been very much for internal use only. So apologies if I run roughshod over some standard conventions/requirements, I will try not to be rude.

I generally like this and really only have a single issue, which is with the label `Changes`, to me this is like a catch-all and too broad a label to be relevant to us in reviewing a pull request.

Comment (marc-vdm) [reply ci/changelog-configuration.json:3]: @haasad FYI: @Zoophobus is the new AB developer who recently started with us. We'll be hearing more from him :)

**Review by Zoophobus — APPROVED** [review]
Whether or not I find some tags redundant is irrelevant to this PR. This is good work and will help make things a lot easier. Thanks Adrian and sorry for delaying on approving this.

**Pull request merged**
